### PR TITLE
[1.16] Fix server restore to not remove podman containers

### PR DESCRIPTION
**Pull request by saschagrunert**
The previous behavior was that CRI-O forcely tried to remove running
podman containers on startup, which only succeeded for the container
storage but not the running process. The result was was a running
container process in podman, which got broken because its storage got
removed.

This is now fixed by excluding podman containers on restore, discovered
by the new podman annotation.

A dedicated unit test has been added as well.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,7 +16,7 @@ require (
 	github.com/containers/buildah v1.11.2
 	github.com/containers/conmon v2.0.0+incompatible
 	github.com/containers/image v3.0.2+incompatible
-	github.com/containers/libpod v1.5.1
+	github.com/containers/libpod v1.6.1
```
Comment (haircommander) [inline go.mod]: why the libpod bump?

Comment (saschagrunert) [reply go.mod]: For the annotation to be available, which has been introduced in 1.6.0.

File: internal/lib/container_server.go [hunk 2]
```diff
@@ -441,6 +452,12 @@ func (c *ContainerServer) LoadContainer(id string) error {
 	if err := json.Unmarshal(config, &m); err != nil {
 		return err
 	}
+
+	// Do not interact with containers of others
+	if manager, ok := m.Annotations[annotations.ContainerManager]; ok && manager != ContainerManagerCRIO {
+		return ErrIsNonCrioContainer
+	}
+
```
Comment (TomSweeneyRedHat) [inline internal/lib/container_server.go]: in #2882 there was a check like:

```
return md.PodName != "" && md.PodID != ""
```

Is that worth doing here too?

File: internal/lib/container_server.go [hunk 3]
```diff
@@ -431,6 +440,8 @@ func configNetNsPath(spec *rspec.Spec) (string, error) {
 	return "", fmt.Errorf("missing networking namespace")
 }
 
+var ErrIsNonCrioContainer = errors.New("non CRI-O container")
```
Comment (TomSweeneyRedHat) [inline internal/lib/container_server.go]: I thought there was a file where Crio errors were defined?  Should this be there or am I'm having a bad memory?

File: internal/lib/container_server.go (hunk 2 above)
Comment (haircommander) [reply internal/lib/container_server.go]: Yeah I think doing both. If we're restoring from an older version we still want to restore our crio containers that don't yet have it. We can deprecate the Pod{Name,ID} check in 1.17

File: go.mod (hunk 1 above)
Comment (haircommander) [reply go.mod]: right silly me :upside_down_face:

File: internal/lib/container_server.go (hunk 2 above)
Comment (saschagrunert) [reply internal/lib/container_server.go]: I agree, I added it inspired by https://github.com/cri-o/cri-o/pull/2882

File: internal/lib/container_server.go (hunk 3 above)
Comment (saschagrunert) [reply internal/lib/container_server.go]: Hm, we don't have that many dedicated errors, so I decided to keep the error nearby where it's actually used. Do you think that we should create a dedicated place for errors?

Comment (TomSweeneyRedHat) [reply internal/lib/container_server.go]: If we don't have a dedicated error file, then don't worry about it for now.  However, I personally do prefer having a central common file for hard error definitions like this one.   Maybe something to consider for a follow up PR if @mrunalp is hip with the thought.

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests

**Comment by haircommander** [comment]
Unfortunately, this issue is due to our CI infrastructure changing ssh keys, but not updating the authorized_keys files on new instances. I have contacted the maintainer of said keys but haven't heard anything yet..

**Comment by saschagrunert** [comment]
@haircommander It looks like that the CI got broken because of invalid permissions:
```
########## STARTING STAGE: FORWARD PARAMETERS TO THE REMOTE HOST ##########
+ [[ -s /var/lib/jenkins/jobs/test_pull_request_crio_critest_fedora/workspace/activate ]]
+ source /var/lib/jenkins/jobs/test_pull_request_crio_critest_fedora/workspace/activate
++ export VIRTUAL_ENV=/var/lib/jenkins/origin-ci-tool/e7531a43acf671d647d0c19b3d1fddbc79373006
++ VIRTUAL_ENV=/var/lib/jenkins/origin-ci-tool/e7531a43acf671d647d0c19b3d1fddbc79373006
++ export PATH=/var/lib/jenkins/origin-ci-tool/e7531a43acf671d647d0c19b3d1fddbc79373006/bin:/sbin:/usr/sbin:/bin:/usr/bin
++ PATH=/var/lib/jenkins/origin-ci-tool/e7531a43acf671d647d0c19b3d1fddbc79373006/bin:/sbin:/usr/sbin:/bin:/usr/bin
++ unset PYTHON_HOME
++ export OCT_CONFIG_HOME=/var/lib/jenkins/jobs/test_pull_request_crio_critest_fedora/workspace/.config
++ OCT_CONFIG_HOME=/var/lib/jenkins/jobs/test_pull_request_crio_critest_fedora/workspace/.config
+ ssh -F /var/lib/jenkins/jobs/test_pull_request_crio_critest_fedora/workspace/.config/origin-ci-tool/inventory/.ssh_config openshiftdevel sudo chmod o+rw /etc/environment
Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).
++ export status=FAILURE
++ status=FAILURE
```
Do you think we can fix this somehow?

**Comment by umohnani8** [comment]
LGTM

**Comment by haircommander** [comment]
replaced by #2890

**Pull request closed without merging**